classes/local/redaction_method.php

```diff
@@ -0,0 +1,38 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * The redaction method interface describes the required methods for each redaction strategy
+ *
+ * @package   tool_fileredact
+ * @author    Kevin Pham <kevinpham@catalyst-au.net>
+ * @copyright Catalyst IT, 2022
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace tool_fileredact\local;
+
+interface redaction_method {
+
+    /**
+     * Processes redaction for the given file.
+     *
+     * @param \stdClass $filerecord
+     * @param array $hookargs
+     * @return bool as to whether the operation was successful
+     */
+    public function run(\stdClass $filerecord, array $hookargs): bool;
```

Comment:
This is fine, just feels like slight overkill

Reply:
It does at first, yeah.

But I believe this will make it easier to add new redaction methods going forward, and keeps the caller of the implementation pretty simple. The one thing I'm unsure of at this point, is where those classes "live". But for now I've grouped them under their respective file type targets (doesn't necessarily have to be though).